iox_time/src/lib.rs

```diff
@@ -179,6 +182,12 @@ impl Time {
 pub trait TimeProvider: std::fmt::Debug + Send + Sync + 'static {
     /// Returns the current `Time`. No guarantees are made about monotonicity
     fn now(&self) -> Time;
+
+    /// Sleep for the given duration.
+    fn sleep(&self, d: Duration) -> Pin<Box<dyn Future<Output = ()> + Send + 'static>>;
```

Comment:
I wonder if we could provide a default implementation here

```suggestion
    fn sleep(&self, d: Duration) -> Pin<Box<dyn Future<Output = ()> + Send + 'static>> {
        self.sleep_until(self.now() + d)
     }
```

and we might be able to cut down the code in this PR by half 🤔

Reply:
I've applied your suggestion but left the tests because I thought it easier to have both functions tested than to reason about "which subset / semantic do I need to test under which circumstances".